### PR TITLE
v1alpha4 support

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:

v1alpha4 support

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @wentao-zh 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
v1alpha4 support
```

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -6,29 +6,6 @@ metadata:
   creationTimestamp: null
   name: manager-role
 rules:
-- apiGroups:
```
Comment (gyliu513) [inline config/rbac/role.yaml:9]: @mkumatag so we do not need those rules anymore for secrets and events?

**Review by gyliu513 — COMMENTED** [review]
Thanks @mkumatag , looks great and just one minor comment want to get your conform for RBAC.

File: config/rbac/role.yaml (hunk 1 above)
Comment (mkumatag) [reply config/rbac/role.yaml:9]: na! I think this is happening part of autogenerated code, lemme fix the annotations in the controller to keep these rules intact.

Comment (mkumatag) [reply config/rbac/role.yaml]: @gyliu513 added the kubebuilder rule to the controller with the required permissions. ptal at latest commit

**Comment by gyliu513** [comment]
/lgtm

Thanks @mkumatag , looks good to me.

@wentao-zh can you help get someone to do some test against this new PR?

**Comment by wentao-zh** [comment]
> /lgtm
> 
> Thanks @mkumatag , looks good to me.
> 
> @wentao-zh can you help get someone to do some test against this new PR?

Yes. I'm working on this testing right now.

**Comment by gyliu513** [comment]
@wentao-zh @mkumatag since cluster api has some break changes between different versions as it is still involving fast, can we merge this PR? I think we do not need to support cluster api 0.3.0 as long as the current branch works after the PR got merged, comments?

**Comment by mkumatag** [comment]
> @wentao-zh @mkumatag since cluster api has some break changes between different versions as it is still involving fast, can we merge this PR? I think we do not need to support cluster api 0.3.0 as long as the current branch works after the PR got merged, comments?

agree,!

**Comment by jichenjc** [comment]
+1 , unless it become better, the API will change very fast, if we really need think about some kind of alpah3 to alpha4, maybe can refer to 
https://cluster-api.sigs.k8s.io/developer/providers/v1alpha3-to-v1alpha4.html
but guess not necessary

**Comment by mkumatag** [comment]
> +1 , unless it become better, the API will change very fast, if we really need think about some kind of alpah3 to alpha4, maybe can refer to
> https://cluster-api.sigs.k8s.io/developer/providers/v1alpha3-to-v1alpha4.html
> but guess not necessary

I have used this guide and adopted the recommendation wherever necessary,

**Pull request merged**